lms/views/api/blackboard/_schemas.py

```diff
@@ -31,7 +32,15 @@ class BlackboardListFilesSchema(RequestsResponseSchema):
 
     @post_load
     def post_load(self, data, **_kwargs):  # pylint:disable=no-self-use
-        return data["results"]
+        pdf_files = []
+
+        for file in data["results"]:
+            if file.get("mime_type") == "application/pdf":
+                # Delete mime_type: we don't want to send it to the frontend.
+                del file["mime_type"]
+                pdf_files.append(file)
+
+        return pdf_files
```

Comment:
Following the existing design here `BlackboardListFilesSchema` is responsible for converting the files into the format expected by our frontend: renaming fields, omitting fields that we don't use, etc. `BlackboardListFilesSchema` and `TestBlackboardListFilesSchema` are the only places in our code where knowledge of the Blackboard API response format exists. The view just treats the list of dicts returned by `BlackboardListFilesSchema` as opaque and blindly returns them to the frontend.

At least for now this seems to be the best design. In earlier PRs we tried a version where the schema just validates the responses and doesn't mutate them at all, and the view does the mutation. But it just made the code and tests longer and spread knowledge of the Blackboard API response format around.

I think this could likely change in future as requirements change

Reply:
> I think this could likely change in future as the code design changes in response to changing requirements

Yeah I've seen the changes for folders and even If I'm not a huge fan of the schema doing a lot of work this is simple enough to be easily change later on.